csharp/src/Ice/OutputStream.cs

```diff
@@ -1137,6 +1158,27 @@ private static int Distance(IList<ArraySegment<byte>> data, Position start, Posi
             return size + end.Offset;
         }
 
+        /// <summary>Writes a size into a span of bytes using a fixed number of bytes (currently always 4 bytes).
+        /// </summary>
+        /// <param name="size">The size to write.</param>
+        /// <param name="data">The destination byte buffer.</param>
+        private static void WriteFixedLengthSize(int size, Span<byte> data)
+        {
+            if (size < 0 || size > 1_073_741_823) // 2^30 -1
+            {
+                throw new ArgumentOutOfRangeException("size is out of range", nameof(size));
+            }
+            if (data.Length < 4)
```

Comment:
Should we replace this with an Assert, given that this is only used internally

Reply:
Or nothing at all since `MemoryMarshal.Write` fails if the buffer is too small.